src/freenet/client/filter/TheoraPacketFilter.java

```diff
@@ -57,17 +57,17 @@ private void verifyIdentificationHeader(BitInputStream input) throws IOException
 
         checkHeaderField("Identification", "VMAJ", input, 8, not(v -> v != 3));
         checkHeaderField("Identification", "VMIN", input, 8, not(v -> v != 2));
-        checkHeaderField("Identification", "VREV", input, 8, not(v -> v > 1));
-        int FMBW = checkHeaderField("Identification", "FMBW", input, 16, not(v -> v == 0));
-        int FMBH = checkHeaderField("Identification", "FMBH", input, 16, not(v -> v == 0));
+        input.skip(8); // skip VREV
+        int FMBW = checkHeaderField("Identification", "FMBW", input, 16, v -> v > 0);
+        int FMBH = checkHeaderField("Identification", "FMBH", input, 16, v -> v > 0);
         checkHeaderField( "Identification", "PICW", input, 24, not(v -> v > FMBW * 16));
         checkHeaderField("Identification", "PICH", input, 24, not(v -> v > FMBH * 16));
         checkHeaderField("Identification", "PICX", input, 8, not(v -> v > FMBW * 16 - v));
```

Comment:
Hmm, is this not the same as `2 * v > FMBW * 16` which would be the same as `v > FMBW * 8`?

Reply:
This is the same. And where does this come from:

> Read an 8-bit unsigned integer as PICX.  This MUST be no greater than (FMBW∗16−PICX).

Should it be simplified?

Reply:
No, doesn’t have to be. But there’s an unnecessary space two lines above so if you could run the formatter on anything you’ve changed? Also, for Java 8 you’d need to pull in next so the Travis build will run.

Reply:
I already merged next here. Probably these lines should be changed:
https://github.com/freenet/fred/blob/next/build.gradle#L58

Reply:
Those lines were changed in next but you botched the merge, changing them back. :)